File: app/src/main/java/nl/jpelgrm/movienotifier/data/AccountListDiffCallback.java
package nl.jpelgrm.movienotifier.data;

import android.support.v7.util.DiffUtil;

import java.util.List;

import nl.jpelgrm.movienotifier.models.User;

public class AccountListDiffCallback extends DiffUtil.Callback {
    private List<User> oldList;
    private List<User> newList;
    private String oldActive;
    private String newActive;

    public AccountListDiffCallback(List<User> oldList, List<User> newList, String oldActive, String newActive) {
        this.oldList = oldList;
        this.newList = newList;
        this.oldActive = oldActive;
        this.newActive = newActive;
    }

    @Override
    public int getOldListSize() {
        return oldList.size();
    }

    @Override
    public int getNewListSize() {
        return newList.size();
    }

    @Override
    public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {
        return oldList.get(oldItemPosition).getID().equals(newList.get(newItemPosition).getID());
    }

    @Override
    public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
        User oldUser = oldList.get(oldItemPosition);
        User newUser = newList.get(newItemPosition);

        return oldUser != null && newUser != null && oldUser.getID().equals(newUser.getID()) && oldUser.getName().equals(newUser.getName())
                && oldUser.getEmail().equals(newUser.getEmail()) && oldUser.getNotifications().equals(newUser.getNotifications())
                && oldUser.getApikey().equals(newUser.getApikey()) && (oldActive.equals(newActive) && (oldActive.equals(oldUser.getID()) || newActive.equals(newUser.getID())));
    }
}

File: app/src/main/java/nl/jpelgrm/movienotifier/data/AccountListDiffCallback.kt
package nl.jpelgrm.movienotifier.data

import androidx.recyclerview.widget.DiffUtil
import nl.jpelgrm.movienotifier.models.User

class AccountListDiffCallback(private val oldList: List<User>, private val newList: List<User>, private val oldActive: String, private val newActive: String) : DiffUtil.Callback() {
    override fun getOldListSize(): Int {
        return oldList.size
    }

    override fun getNewListSize(): Int {
        return newList.size
    }

    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return oldList[oldItemPosition].id == newList[newItemPosition].id
    }

    override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        val oldUser = oldList[oldItemPosition]
        val newUser = newList[newItemPosition]
        return oldUser.id == newUser.id && oldUser.name == newUser.name
                && oldUser.email == newUser.email && oldUser.apikey == newUser.apikey && oldActive == newActive
                && (oldActive == oldUser.id || newActive == newUser.id)
    }
}